### ISSUE TITLE
dbug_proxy! binding for method which returns an array of object paths

### ISSUE THREAD

**Issue by zeenix** [dbus2/zbus#332]
In GitLab by @danburkert on May 6, 2023, 20:06

The `dbug_proxy!` macro provides the `object` annotation generating a proxy return type for methods which return an object path - it would be nice if there were an equivalent annotation for handling methods that return an array of object paths.

**Comment by bilelmoussaoui** [comment]
I have also seen `HashMap<SomeTyoe, ObjectPath>` on the wild

**Comment by zeenix** [comment]
> I have also seen `HashMap<SomeTyoe, ObjectPath>` on the wild

right. I think we've to draw the line somewhere in terms of macro-provided functionality. :)

**Comment by yakumoklesk** [comment]
This means that when an interface, like for example org.freedesktop.network1.Manager, returns a value like (a{iso}) for the specific Links found, we cannot use object attribute for getting LinkProxys instantiated? The workaround I've found is just to ignore the object attribute and build each LinkProxy given the returned object_path. Is far from elegant though.

**Comment by zeenix** [comment]
> we cannot use object attribute for getting LinkProxys instantiated? The workaround I've found is just to ignore the object attribute and build each LinkProxy given the returned object_path. Is far from elegant though.

As I wrote before here, we can't possibly make everything super convenient but there is a reason this issue is open: someone needs to make it happen. That could be you.
